test/protocol_test.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	rpc "github.com/keybase/go-framed-msgpack-rpc"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/net/context"
+)
+
+var testPort int = 8089
+
+func TestMain(m *testing.M) {
+	err := prepServer()
+	if err != nil {
+		fmt.Println("A server error occurred")
+		os.Exit(-1)
+	}
+	os.Exit(m.Run())
+}
+
+func prepServer() error {
+	server := &Server{port: testPort}
+
+	serverReady := make(chan struct{})
+	var err error
+	go func() {
+		err = server.Run(serverReady)
+	}()
+	<-serverReady
+	return err
+}
+
+func prepClient(t *testing.T) TestClient {
+	c, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", testPort))
+	assert.Nil(t, err, "a dialer error occurred")
+
+	xp := rpc.NewTransport(c, nil, nil)
+	return TestClient{GenericClient: rpc.NewClient(xp, nil)}
+}
+
+func TestCall(t *testing.T) {
+	cli := prepClient(t)
+
+	B := 34
+	for A := 10; A < 23; A += 2 {
+		res, err := cli.Add(context.Background(), AddArgs{A: A, B: B})
+		assert.Nil(t, err, "an error occurred while adding parameters")
+		assert.Equal(t, A+B, res, "Result should be the two parameters added together")
+	}
+}
+
+func TestBrokenCall(t *testing.T) {
+	cli := prepClient(t)
+
+	err := cli.Broken()
+	assert.Error(t, err, "Called nonexistent method, expected error")
+}
+
+func TestNotify(t *testing.T) {
+	cli := prepClient(t)
+
+	pi := 31415
+
+	err := cli.UpdateConstants(context.Background(), Constants{Pi: pi})
+	assert.Nil(t, err, "Unexpected error on notify: %v", err)
+
+	time.Sleep(3 * time.Millisecond)
```

Comment:
can remove the flakiness here same as in test/client.go (with channels), or have a TODO